Common ques
1. can we use lab? YES
2. Can we use Jupyter notebook locally? YES
3. Can we use neuro lab? YES
4. Can we use VSCODE? YES
4. Can we use PyCharm? YES
5. Can we use Colab? YES
6. Can we code simultaneosly code? YES definitely

# AGENDA
1. Variable Case senstivity [clear]
2. Reserved words in Python [clear]
3. Multi assignment of variables [clear]
4. User input [clear]
5. type casting | type conversion [Doubt at the end]
6. Memory block []
7. Operators
  - Logical
  - Equality
  - Comparision
  - Arthimatic
  - Bitwise [Advance classes]
  - Operators for sets and dictionary [Advance classes]

8. String methods

## Variable Case senstivity

variable names defined as

Ashish ashish

In [ ]:
TEN = 10
ten = 20
# TEN and ten both are different in terms of lowercase and uppercase letter

In [ ]:
print(Ten)

NameError: ignored

In [ ]:
TEN

10

## Reserved words in Python

https://docs.python.org/2.5/ref/keywords.html

1. We should not use reserved words as variable names
2. Also python doesn't restrict the user to use reserved keywords as variable names

In [ ]:
print("Hi there")

Hi there


In [ ]:
print = "Hi there" # as a variable
type(print)

str

In [ ]:
print("Hi there")

TypeError: ignored

In [ ]:
print = 12 # as a variable
type(print)

int

In [ ]:
print("Hi there")

TypeError: ignored

In [ ]:
del print # delete the variable print

In [ ]:
print("Hi there")

Hi there


In [ ]:
Print = "Hi there" # as a variable
type(Print)

str

In [ ]:
print("Hi there")

Hi there


In [ ]:
print = 10 

In [ ]:
type(print)

int

In [ ]:
del print

In [ ]:
print("Sunny")

Sunny


## Multi assignment of variables

In [ ]:
a, b, c, yes = 10, 20.2, "Sunny", True

In [ ]:
type(a)

int

In [ ]:
type(b)

float

In [ ]:
type(c)

str

In [ ]:
type(yes)

bool

In [ ]:
type(a)

int

In [ ]:
print(type(a))
print(type(b))
print(type(c))
print(type(yes))

<class 'int'>
<class 'float'>
<class 'str'>
<class 'bool'>


In [ ]:
# a,b,c = 10

In [ ]:
a,b,c = 10,10,10

In [ ]:
a, A, c, C = 10, 20.2, "Sunny", True

In [ ]:
print(a, A, c, C)

10 20.2 Sunny True


In [ ]:
a=b=c=10

In [ ]:
a,b,c

(10, 10, 10)

In [ ]:
firstName, middleName, sirName = "Sunny", "Bhaveen", "Chandra"

In [ ]:
print(f"{firstName}.{sirName}@gmail.com")

Sunny.Chandra@gmail.com


In [ ]:
name, place, year_of_birth, weight = "Sunny", "Bengaluru", 1991, 80.88

In [ ]:
print(f"""
My name is: {name}
I live at: {place}
I was born in: {year_of_birth}
My weight is: {weight} in Kgs
""")


My name is: Sunny
I live at: Bengaluru
I was born in: 1991
My weight is: 80.88 in Kgs



# NOTE: Before can i use? Please try in colab or jupyter notebook once

## User input

In [ ]:
name = input("Enter your Name: ")

print(f"You have entered: {name}")

Enter your Name: 1234
You have entered: 1234


In [ ]:
A = input("Enter value of A: ")
B = input("Enter value of B: ")

print(f"Sum of A and B is: {A+B}")

Enter value of A: 12
Enter value of B: 10
Sum of A and B is: 1210


In [ ]:
print("Sum of A and B is: {A+B}")

Sum of A and B is: {A+B}


In [ ]:
A = input("Enter value of A: ")
B = input("Enter value of B: ")

print(type(A), type(B))

Enter value of A: 12
Enter value of B: 10
<class 'str'> <class 'str'>


In [ ]:
name = input("Name: ")
place = input("place: ")
year_of_birth = input("year_of_birth: ")



In [ ]:
print(f"""
My name is: {name}
I live at: {place}
I was born in: {year_of_birth}
""")


My name is: Bhaveen
I live at: Lucknow
I was born in: 1996



In [ ]:
# calcualte the age of the user

name = input("Name: ") # always gives you string values
place = input("place: ")
year_of_birth = input("year_of_birth: ")

age = 2022 - year_of_birth

print(f"""
My name is: {name}
I live at: {place}
{name} is of age: {age}
""")

Name: Sunny
place: BLR
year_of_birth: 1991


TypeError: ignored

In [ ]:
# calcualte the age of the user

name = input("Name: ") # always gives you string values
place = input("place: ")
year_of_birth = int(input("year_of_birth: ")) # convert this into integer

age = 2022 - year_of_birth

print(f"""
My name is: {name}
I live at: {place}
{name} is of age: {age}
""")

Name: Sunny
place: BLR
year_of_birth: 1991

My name is: Sunny
I live at: BLR
Sunny is of age: 31



In [ ]:
# calcualte the age of the user

name = input("Name: ") # always gives you string values
place = input("place: ")
year_of_birth = input("year_of_birth: ") # convert this into integer

age = 2022 - int(year_of_birth)

print(f"""
My name is: {name}
I live at: {place}
{name} is of age: {age}
""")

Name: Sunny
place: BLR
year_of_birth: 1991

My name is: Sunny
I live at: BLR
Sunny is of age: 31



In [ ]:
# calcualte the age of the user

name = input("Name: ") # always gives you string values
place = input("place: ")
year_of_birth = input("year_of_birth: ") # convert this into integer
current_year = input("current year: ")
age = int(current_year) - int(year_of_birth)

print(f"""
My name is: {name}
I live at: {place}
{name} is of age: {age}
""")

Name: Sunny
place: BLR
year_of_birth: 1991
current year: 2023

My name is: Sunny
I live at: BLR
Sunny is of age: 32



## Type conversion

We can convert the data type of one variable into another if it is allowed


In [ ]:
A = int(input("Enter value of A: "))
B = int(input("Enter value of B: "))

print(type(A), type(B))

Enter value of A: 12
Enter value of B: 10
<class 'int'> <class 'int'>


In [ ]:
print(f"Sum of A and B is: {A+B}")

Sum of A and B is: 22


In [ ]:
int A

SyntaxError: ignored

In [1]:
float(input(""))

3


3.0

In [ ]:
A = float(input("Enter value of A: "))
B = float(input("Enter value of B: "))

print(f"Sum of A and B is: {A+B}")

Enter value of A: 12.2
Enter value of B: 10.2
Sum of A and B is: 22.4


In [ ]:
A = float(input("Enter value of A: "))
B = int(input("Enter value of B: "))

print(f"Sum of A and B is: {A+B}")

Enter value of A: 12.2
Enter value of B: 10
Sum of A and B is: 22.2


In [ ]:
A = float(input("Enter value of A: "))
B = int(input("Enter value of B: ")) # 10.2 is float but converting to int

print(f"Sum of A and B is: {A+B}")

Enter value of A: 10.2
Enter value of B: 10.2


ValueError: ignored

In [ ]:
# String to integer
A = "50"
print(f"data type before: {type(A)}, value of A: {A}")
A = int(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'str'>, value of A: 50
data type after: <class 'int'>, value of A: 50


In [ ]:
# String to integer
A = "50.22" # entering float
print(f"data type before: {type(A)}, value of A: {A}")
A = int(A) # trying to convert into integer
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'str'>, value of A: 50.22


ValueError: ignored

In [ ]:
# float to integer
A = 50.22
print(f"data type before: {type(A)}, value of A: {A}")
A = int(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'float'>, value of A: 50.22
data type after: <class 'int'>, value of A: 50


In [ ]:
# integer to float
A = 50
print(f"data type before: {type(A)}, value of A: {A}")
A = float(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'int'>, value of A: 50
data type after: <class 'float'>, value of A: 50.0


In [14]:
bool(-4)

True

In [ ]:
# integer to boolean
A = 10
print(f"data type before: {type(A)}, value of A: {A}")
A = bool(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'int'>, value of A: 10
data type after: <class 'bool'>, value of A: True


* True in binary - 1 
* False in binary - 0 

In [ ]:
# integer to boolean
A = 0
print(f"data type before: {type(A)}, value of A: {A}")
A = bool(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'int'>, value of A: 0
data type after: <class 'bool'>, value of A: False


In [ ]:
# integer to boolean
A = 1
print(f"data type before: {type(A)}, value of A: {A}")
A = bool(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'int'>, value of A: 1
data type after: <class 'bool'>, value of A: True


> NOTE: Converting any non zero number and string values into boolean will give True as an answer

> NOTE: Empty string "" will give False as an answer after converting from string to boolean

In [ ]:
# integer to boolean
A = -5
print(f"data type before: {type(A)}, value of A: {A}")
A = bool(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'int'>, value of A: -5
data type after: <class 'bool'>, value of A: True


In [ ]:
# string to boolean
A = "Sunny"
print(f"data type before: {type(A)}, value of A: {A}")
A = bool(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'str'>, value of A: Sunny
data type after: <class 'bool'>, value of A: True


In [ ]:
# string to boolean
A = "True"
print(f"data type before: {type(A)}, value of A: {A}")
A = bool(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'str'>, value of A: True
data type after: <class 'bool'>, value of A: True


In [ ]:
# string to boolean
A = "False"
print(f"data type before: {type(A)}, value of A: {A}")
A = bool(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'str'>, value of A: False
data type after: <class 'bool'>, value of A: True


In [ ]:
# empty string to boolean
A = ""
print(f"data type before: {type(A)}, value of A: {A}")
A = bool(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'str'>, value of A: 
data type after: <class 'bool'>, value of A: False


In [17]:
bool(" ")

True

In [ ]:
# empty string to boolean
A = " "
print(f"data type before: {type(A)}, value of A: {A}")
A = bool(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'str'>, value of A:  
data type after: <class 'bool'>, value of A: True


In [ ]:
# empty string to int
A = ""
print(f"data type before: {type(A)}, value of A: {A}")
A = int(A)
print(f"data type after: {type(A)}, value of A: {A}")

data type before: <class 'str'>, value of A: 


ValueError: ignored

## Memory block 

In [ ]:
a = 22
b = 34
c = 22 

In [ ]:
id(a) # memory address

11123264

In [ ]:
id(b)

11123648

In [ ]:
id(c)

11123264

In [ ]:
c = 22 + b


In [ ]:
id(c)

11124352

In [ ]:
a = 22
id(a)

11123264

In [ ]:
a = 99
id(a)

11125728

Memory is assigned to the value of variable passed not the variable name

In [ ]:
a = 22
b = 34
c = 22 

print(id(a), id(b), id(c))

11123264 11123648 11123264


In [ ]:
a = "Sunny"
b = 34
c = "Sunny"

print(id(a), id(b), id(c))

140359210811888 11123648 140359210811888


In [ ]:
a = "S"
b = 34
c = "S"

print(id(a), id(b), id(c))

140359751722032 11123648 140359751722032


In [ ]:
a = 22
b = 34
c = 22.0 

print(id(a), id(b), id(c))

11123264 11123648 140359209645776


# LIVE DOUBTS

In [ ]:
name = input("Name: ")
YOB = int(input("YOB: ")) # string

age = 2022 - YOB
print(age)


Name: Sunny
YOB: 1991
31


# ASSIGNMENTS - NOT compulsory but usefull!!
# I am not at loss if you don't submit
## send the mail to 
## send it before 0000 hrs IST | 12 AM tonight
## Special cases respond atleast before 1200 hrs IST tomorrow i.e Monday

TO: query@ineuron.ai
CC: sunny@ineuron.ai, riishav@ineuron.ai, boktiar@ineuron.ai
 
Subject: [Bootcamp 2.0 | Assignment No. 1]: 

1. Find out all the types of error that we have encountered in our lecture and why? Also include the examples.
2. Take input A and B and print their sum, mul, div, square of the nos.
3. How many reserve keywords are there in python and why we should not use them as a variable name.
4. Create email ids with the user inputs. in the following format - 
lastname.firstname@gmail.com
5. Try to print the following lines - 
  - Hi I'm Sunny
  - My address is 22\3 
  - I earn 20$ daily.
6. Take user input (name, address, date of birth etc) to fill a form for your college and print their email IDs
7. Convert the following - 
  - "22.4" into integer
  - "22" into complex no.


In [ ]:
x = input("Enter the value of x: ")

Enter the value of x: 112


In [ ]:
x =  22.2
bool(x)

True

In [ ]:
x =  22
bool(x)

True

In [ ]:
x =  0.0
bool(x)

False

In [ ]:
x =  0
bool(x)

False

In [26]:
bool(7)

True

In [ ]:
x =  -22
bool(x)

True

In [ ]:
x =  -22.2
bool(x)

True

In [ ]:
i_have_a_doubt = input("yes? ")
bool(i_have_a_doubt)

yes? yes


True

In [27]:
i_have_a_doubt = input("yes? ")
bool(i_have_a_doubt)

yes? yes


True

In [ ]:
25.88 + 10

35.879999999999995

In [ ]:
round(25.88 + 10, 2)

35.88

In [ ]:
25.89 + 10

35.89

In [ ]:
def sum_of_values(x, y):
  print(x+y)

sum_of_values(22.2, 33, 444)

TypeError: ignored